docs/src/main/asciidoc/mailer-reference.adoc

```diff
@@ -283,8 +283,64 @@ Check the xref:vertx.adoc[Using Vert.x guide] for further details about these di
 The retrieved `MailClient` is configured using the configuration key presented above.
 You can also create your own instance, and pass your own configuration.
 
+
+== Using SSL with native executables
+
+Note that if you enable SSL for the mailer and you want to build a native executable, you will need to enable the SSL support.
+Please refer to the xref:native-and-ssl.adoc[Using SSL With Native Executables] guide for more information.
+
+== Configuring the SMTP credentials
+
+It is recommended to encrypt any sensitive data, such as the `quarkus.mailer.password`.
+One approach is to save the value into a secure store like HashiCorp Vault, and refer to it from the configuration.
```

Comment:
Add a link to the vault extension guide?

Reply:
Hum, didn't change that part, but yes, let's add a link to the vault extension.